### PR TITLE
Add code-analysis pipeline with Whitesource scan

### PR DESCRIPTION
**Pull request by metacubed**

### CODE REVIEW

File: AnalysisJenkinsfile [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                                    echo 'Downloading Propel OSS binary'
+                                    wget --http-user=${ARTIFACTORY_READER} \
+                                         --http-password=${ARTIFACTORY_READER_PW} \
+                                         https://artifactory.build.ge.com/BUILDGE/propel-oss/latest/linux/propel-oss
+                                    chmod u+x propel-oss
+    
+                                    echo 'Running scan'
+                                    ./propel-oss
+                                '''
+                            }
```
Comment (irinaepshteyn) [inline AnalysisJenkinsfile:85]: Is there any report to analyze/download? Should we fail the job if license issues or vulnerabilities are detected?

File: AnalysisJenkinsfile [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        stage ('Run code analysis') {
+            parallel {
+                stage ('Whitesource') {
+                    when {
+                        beforeAgent true
+                        expression { params.WHITESOURCE_SCAN }
+                    }
+                    agent {
+                        docker {
+                            image 'openjdk:8'
```
Comment (irinaepshteyn) [inline AnalysisJenkinsfile:25]: Have you looked at examples at https://propel.ci.build.ge.com/docs/oss-scan/#propel-cli?  Image they are using in the examples: registry.gear.ge.com/dig-propel/propel-oss:latest.
Is it better to use so you don't have to download oss binary?

File: build.gradle [hunk 3]
```diff
@@ -211,6 +211,14 @@ subprojects {
             csv.enabled = false
         }
     }
+
+    if (!path.startsWith(':cloudfoundry-identity-samples')) {
+        task collectDependencies(type: Copy) {
+            from configurations.default
+            into "${rootProject.buildDir}/dependencies"
+            exclude 'cloudfoundry-identity-*.jar'
```
Comment (irinaepshteyn) [inline build.gradle:219]: Do we really need to exclude uaa jars from dependencies? Are they really included otherwise?

File: build.gradle [hunk 4]
```diff
@@ -211,6 +211,14 @@ subprojects {
             csv.enabled = false
         }
     }
+
+    if (!path.startsWith(':cloudfoundry-identity-samples')) {
+        task collectDependencies(type: Copy) {
+            from configurations.default
```
Comment (irinaepshteyn) [inline build.gradle:217]: What is gradle default configuration? We only need runtime for collecting dependencies, right?

**Review by irinaepshteyn — COMMENTED** [review]
@metacubed, done reviewing.  Please address my comments.

File: AnalysisJenkinsfile [hunk 5]
```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent none
+    environment {
+        COMPLIANCEENABLED = true
```
[the remark this answers is not in the document]
Comment (metacubed) [reply AnalysisJenkinsfile:6]: I'm pretty sure the flag doesn't do _anything_. Not sure though, and the build samples use it everywhere. I'll remove it for now.

File: AnalysisJenkinsfile [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    }
+    options {
+        skipDefaultCheckout()
+        buildDiscarder(logRotator(artifactDaysToKeepStr: '1', artifactNumToKeepStr: '1', daysToKeepStr: '5', numToKeepStr: '10'))
+    }
+    parameters {
+        booleanParam(name: 'WHITESOURCE_SCAN', defaultValue: true, description: 'Run Whitesource scan')
```
[the remark this answers is not in the document]
Comment (metacubed) [reply AnalysisJenkinsfile:13]: Yes, I want to add Coverity, and maybe Checkmarx and  FindBugs eventually. That's also why I wrapped it in a `parallel` stage.

File: AnalysisJenkinsfile (hunk 1 above)
Comment (metacubed) [reply AnalysisJenkinsfile:85]: The job is triggered asynchronously, and the results appear on the Propel Reports page on completion.

File: AnalysisJenkinsfile (hunk 2 above)
Comment (metacubed) [reply AnalysisJenkinsfile:25]: Our `build.gradle` requires setting environment variables for the Artifactory creds. That's not possible if we run it using the OSS image. The job internally creates a new VM using the image specs, so the parent env is not carried over.

File: build.gradle (hunk 3 above)
Comment (metacubed) [reply build.gradle:219]: `uaa.war` uses `service.jar`, `model.jar`, etc. So they do show up as dependencies with "Unknown license". That's why I excluded them.

File: build.gradle (hunk 4 above)
Comment (metacubed) [reply build.gradle:217]: Hmm, `default` == `runtime` if not customized. I can try switching to runtime.

File: AnalysisJenkinsfile (hunk 6 above)
[the remark this answers is not in the document]
Comment (irinaepshteyn) [reply AnalysisJenkinsfile:13]: Gotta. Sounds good then

**Review by irinaepshteyn — APPROVED** [review]
Looks good.

**Pull request merged**